tree/ntuple/v7/src/RField.cxx

```diff
@@ -2368,6 +2394,50 @@ void ROOT::Experimental::RVariantField::CommitCluster()
 
 //------------------------------------------------------------------------------
 
+ROOT::Experimental::RSetField::RSetField(std::string_view fieldName, std::unique_ptr<Detail::RFieldBase> itemField)
+   : ROOT::Experimental::RProxiedCollectionField(fieldName, "std::set<" + itemField->GetType() + ">")
+{
+}
+
+std::unique_ptr<ROOT::Experimental::Detail::RFieldBase>
+ROOT::Experimental::RSetField::CloneImpl(std::string_view newName) const
+{
+   auto newItemField = fSubFields[0]->Clone(fSubFields[0]->GetName());
+   return std::make_unique<RSetField>(newName, std::move(newItemField));
+}
+
+void ROOT::Experimental::RSetField::ReadGlobalImpl(NTupleSize_t globalIndex, Detail::RFieldValue *value)
+{
+   if (!fProxy->GetCollectionClass()->HasDictionary()) {
+      throw RException(R__FAIL("RField: no dictionary loaded for collection type " + GetNormalizedType(fProxy->GetCollectionClass()->GetName()).first));
+   }
+
```

Comment:
In principle, this should be evaluated only once; let's move this to the constructor.

Reply:
The reason I initially added it here is because in principle, for the templated version, dictionaries are not needed. But yes, now that you point it out it's not ideal to have to do this check for every entry. I'll move it to the constructor for now but it might be nice to still have the possibility to create templated fields without dictionaries.